src/test/java/org/openrewrite/kotlin/tree/AnnotationTest.java

```diff
@@ -82,4 +84,123 @@ void jvmNameAnnotation() {
+              }
+              """
+          ),
+          // Works
+          kotlin(
+            """
+              import com.example.annotations.JavaAnnotationToRemove
+
+              @JavaAnnotationToRemove
+              class A {
+              }
+              """),
+          // Fails, despite being valid Kotlin code
+          kotlin(
+            """
+              @com.example.annotations.JavaAnnotationToRemove
+              class A {
+              }
+              """)
+        );
+    }
+
+    @Test
+    void fullyQualifieJavaLangAnnotation() {
+        rewriteRun(
+          // Works, Deprecated is defined in java.lang package so no import needed
+          kotlin(
+            """
+              @Deprecated
+              class A {
+              }
+              """),
+          // Fails, despite being valid Kotlin code
+          kotlin(
+            """
+              @java.lang.Deprecated
+              class A {
+              }
+              """)
+        );
+    }
+
+    // Despite being recognized as an annotation, it is not removed (but it works fine with annotations from java.lang package)
+    @Test
+    void customAnnotationIsNotRemoved() {
+        rewriteRun(
+          spec -> spec.recipe(new RemoveAnnotation("@com.example.annotations.JavaAnnotationToRemove")),
+          java(
+            """
+              package com.example.annotations;
+                            
+              import java.lang.annotation.Target;
+              import static java.lang.annotation.ElementType.*;
+                            
+              @Target({TYPE, FIELD, METHOD, PARAMETER, CONSTRUCTOR, LOCAL_VARIABLE})
+              public @interface JavaAnnotationToRemove {
+              }
+              """
+          ),
+          kotlin(
+            """
+              import com.example.annotations.JavaAnnotationToRemove
+                          
+              @JavaAnnotationToRemove
+              class A {
+              }
+              """
+            , """
+              class A {
+              }
+              """)
+        );
+    }
+
+    // Somehow, this one works? Maybe because of the parameters?
+    // Mind the required empty lines after refactor though...
+    @Test
+    void arrayArgumentRemoval() {
+        rewriteRun(
+          spec -> spec.recipe(new RemoveAnnotation("@com.example.Test")),
+          kotlin(
+            """
+              package com.example
+                          
+              @Target ( AnnotationTarget . LOCAL_VARIABLE )
+              @Retention ( AnnotationRetention . SOURCE )
+              annotation class Test ( val values : Array < String > ) {
+              }
+              """
+          ),
+          kotlin(
+            """
+              import com.example.Test
+                          
+              @Test( values = [ "a" , "b" , "c" ] )
+              val a = 42
+              """,
+            """
+              
+              
+              
```

Comment:
Perhaps we'd rather see this fail right? As the excess new lines aren't desirable in the output.
```suggestion
```